### ISSUE TITLE
[Bug] Integration freezes

### ISSUE THREAD

**Issue by man55** [greghesp/ha-bambulab#313]
### Describe the bug

Integration freezes. Often. literally every print. And accordingly, all notifications are broken.
On any version 1.9.x
I rolled back to version 1.7.1 and everything seems to be working again.

This bug already existed a few months ago, then it was fixed, and here it is again.

### To Reproduce

Just update to 1.9.x

### Expected Behaviour

I expected it to work.

### What device are you using?

X1C

### Diagnostic Output

```shell
There should be a log file here. I'm trying to figure out how to add it here.
```

### Other Information

X1C carbon combo Cloud on.
Integration via Local MQTT Connection
[home-assistant_bambu_lab_2023-12-14T20-41-54.661Z.zip](https://github.com/greghesp/ha-bambulab/files/13693943/home-assistant_bambu_lab_2023-12-14T20-41-54.661Z.zip)
[home-assistant_bambu_lab_2023-12-16T15-02-57.392Z.zip](https://github.com/greghesp/ha-bambulab/files/13693944/home-assistant_bambu_lab_2023-12-16T15-02-57.392Z.zip)

**Comment by AdrianGarside** [comment]
Which firmware are you running on your X1? There seems to be a newly introduced bug, similar to the P1P/S behavior, where the printer will stop sending any data. I've seen this affecting both the local mqtt connection and the bambu cloud one (where it affects the handy app as well). They appear to be different bugs.

**Comment by man55** [comment]
latest available firmware 01.07.00
1.7.1 integration works great. The problem with newer version only.

**Comment by AdrianGarside** [comment]
Is there any debug error output in the home assistant logs?

**Comment by Krystm** [comment]
noticed the same issues (on going) since latest firmware release.

**Comment by AdrianGarside** [comment]
> noticed the same issues (on going) since latest firmware release.

Can you provide some debug logs to see if you're hitting the same symptoms.

The one thing that changed in the latest X1 firmware is that it started dumping the entire device state every second again instead of just the delta.

**Comment by AdrianGarside** [comment]
Probable dupe of #256

**Comment by pegoto** [comment]
Just adding this for additional troubleshooting info. The BLLED community, similar LED integration as this, also had an issue with the new firmware. I believe the MQTT message sizes increased which caused their issue. They fixed it in the latest experimental build.

**Comment by AdrianGarside** [comment]
I've made a potential mitigation fix for the symptoms above where we detect we stopped getting data but the previous watchdog behavior of asking nicely to be given data again doesn't seem to be working. Please try:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0-dev2

**Comment by man55** [comment]
Three days later, version 1.7.1 also froze during printing. I'll try v2.0-dev3

**Comment by man55** [comment]
v2.0-dev3 absolutely doesn't work
<img width="536" alt="Screenshot 2023-12-20 122807" src="https://github.com/greghesp/ha-bambulab/assets/30772900/83cbb25e-5cc0-4411-a793-1559c1196569">

**Comment by AdrianGarside** [comment]
Is there more log entry that gives the specific error that bottom line hit?

**Comment by AdrianGarside** [comment]
Never mind. Looks like at some point in the distant past username was optional. This should fix that:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0-dev4

**Comment by AdrianGarside** [comment]
I added more logging locally to try and narrow down the problem and I got a repro. While this doesn't fix the exception (at least I now have a better idea where that's being thrown from), it will fix the mqtt thread hanging permanently on the next update from the X1.
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0-dev5

**Comment by man55** [comment]
Tnx, I'll try v2.0-dev7

**Comment by AdrianGarside** [comment]
@man55 checking in. I'm very confident I've fixed this but wanted to double check with you :)

**Comment by man55** [comment]
Still working. Need to watch for at least three days.

**Comment by AdrianGarside** [comment]
Fixed in v2.0.0.
